js/src/redux/providers/status.js

```diff
@@ -86,10 +118,25 @@ export default class Status {
       secureToken
     };
 
+    return apiStatus;
+  }
+
+  _pollStatus = () => {
+    const nextTimeout = (timeout = 1000) => {
+      if (this._timeoutIds.status) {
+        clearTimeout(this._timeoutIds.status);
+      }
+
+      this._timeoutIds.status = setTimeout(() => this._pollStatus(), timeout);
+    };
+
+    const { isConnected } = this._api;
+    const apiStatus = this.getApiStatus();
+
     const gotConnected = !this._apiStatus.isConnected && apiStatus.isConnected;
 
     if (gotConnected) {
-      this._pollLongStatus();
+      this._pollLongStatus(gotConnected);
```

Comment:
Tiny grumble - not convinced I like the `got` naming here. Would prefer a `is` or `has` especially since this one had me spinning a bit (just not well-known/consistent)